### PR TITLE
Advanced for ranges

### PR DESCRIPTION
**Pull request by tiago4orion**
Enables:

```sh
for i in getfiles("/") {

}
```

and 

```sh
for i in (1 2 3 4 5) {

}
```

### CODE REVIEW

**Comment by vitorarins** [comment]
Awesome! :smile:

File: internal/sh/shell.go [hunk 1]
```diff
@@ -1952,24 +1952,36 @@ func (shell *Shell) executeFor(n *ast.ForNode) (sh.Obj, error) {
 		shell.looping = false
 	}()
 
-	if n.InVar() == "" {
+	if n.InExpr() == nil {
 		return shell.executeInfLoop(n.Tree())
```
Comment (katcipis) [inline internal/sh/shell.go]: for {} rocks :-). One of the first constructs that I just tried out in nash intuitively and it worked :-)

File: internal/sh/shell_test.go [hunk 2]
```diff
@@ -320,6 +320,7 @@ func TestExecuteCmdAssignment(t *testing.T) {
 	}
 }
 
+// IFS *DO NOT* exists anymore. This tests only assure things works as expected (IFS has no power)
```
Comment (katcipis) [inline internal/sh/shell_test.go]: Cant the test name be changed to reflect better the intention, got a little confused on why this has to stay if IFS has no more power.

Comment (tiago4orion) [inline internal/sh/shell_test.go]: yes, makes sense!

**Pull request merged**